### PR TITLE
Reject zero/negative-height detections before track creation in `BYTETracker`

### PR DESCRIPTION
**Pull request by JESUSROYETH**
`STrack.tlwh_to_xyah()` divides by height to build the Kalman `xyah` state:

```python
def tlwh_to_xyah(tlwh: np.ndarray) -> np.ndarray:
    ret = np.asarray(tlwh).copy()
    ret[:2] += ret[2:] / 2
    ret[2] /= ret[3]  # no guard: height 0 -> divide by zero
    return ret
```

A detection with height 0 (or negative) makes that division produce `inf`/`nan`, and that value corrupts the track's `mean`/`covariance` silently: no exception is raised in production, only a numpy `RuntimeWarning` that almost nobody sees. The track stays "alive" with an infinite aspect ratio and keeps diverging from there.

A zero-height detection is not exotic. `clip_boxes()` clamps `y1` and `y2` independently to `[0, h]` without checking `y2 > y1` afterwards, so any prediction with both y-coordinates past the same edge collapses to `y1 == y2`. I confirmed the mechanism against the real function, not a re-implementation, feeding it a box with both y-coordinates below the frame:

```python
>>> clip_boxes(torch.tensor([[300., 490., 350., 520.]]), (480, 640))
tensor([[300., 480., 350., 480.]])
```

I built that box by hand to isolate the mechanism, I did not chase a live model into producing one on a real frame. So I am not claiming a measured rate for how often this happens in practice. What I can say is that `clip_boxes()` itself can and does produce an exact height-0 box, and once it does, the tracker path below has no guard against it.

Glenn flagged this exact bug closing #25594 (a different fix that clamped inside `tlwh_to_xyah` itself) and gave the direction: "Invalid detections must be rejected before track creation, not clamped inside the coordinate conversion."

## Fix

Filtering in `BYTETracker._split_detections()`, the shared owner of the high/low confidence split that `init_track()` reads right before any `STrack` gets built, so the detection never reaches track creation:

```python
wh = (results.xywhr if hasattr(results, "xywhr") else results.xywh)[:, 2:4]
valid = (wh[:, 0] > 0) & (wh[:, 1] > 0)
remain_inds = valid & (scores >= self.args.track_high_thresh)
inds_low = valid & (scores > self.args.track_low_thresh) & (scores < self.args.track_high_thresh)
```

No `BYTETracker` subclass overrides `_split_detections`, so this covers `BYTETracker`, `FASTTracker`, `OCSORT` and `DeepOCSORT` in one place, all of them build on `STrack`/`tlwh_to_xyah`. `BOTSORT` extends `BYTETracker` too and uses `BOTrack.tlwh_to_xywh`, which never divides by height, so it was not broken by this one, it just gets the same filter for free. `TRACKTRACK` is a separate, non-inheriting implementation with its own `update()`, out of scope here. It builds on `BOTrack` as well, so it does not have this specific division bug either. It is just not touched by this patch.

## Verified

Before the fix, `BYTETracker.update()` with one normal box and one degenerate box (`y1 == y2 == 480`, height 0) raises `RuntimeWarning: divide by zero encountered in scalar divide` at `byte_tracker.py:184`, and the surviving track ends up with `mean=[325, 480, inf, 0, ...]`. After the fix, the same input returns only the valid track, no warning, no nan/inf.

I also checked `track.idx` (it points back into the full original detection array, `track.py` uses it later as `predictor.results[i] = result[idx]`) stays correct when the degenerate box is not the last one in the list: degenerate at position 0, valid ones at 1 and 3, output idx still 1 and 3.

Full tracking suite, `pytest tests/test_python.py -k track`: 14 passed, including `test_track_stream` over all 6 trackers on every model in `MODELS` (8 weights, one per task plus a grayscale detect variant) — `test_track_stream` itself skips classify/semantic/depth before calling the tracker, so tracking runs on the 5 models that cover the 4 supported tasks (detect, segment, pose, obb; grayscale is a detect variant), matching the `trackable` tuple in `track.py`.

## Test

No new test added, per repo policy (tests out of scope by default; only an uncovered, high-risk path justifies one, and the fail/pass evidence above already covers that). Deleted: nothing.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Rejects detections with zero or negative width or height before they can create `STrack` instances in `BYTETracker`.

### 📊 Key Changes
- Adds width and height validation in `_split_detections` using `xywhr` when available, otherwise `xywh`.
- Applies the validity mask to both high-confidence and low-confidence detection groups.
- Prevents degenerate boxes from reaching `tlwh_to_xyah` and corrupting Kalman filter state.

### 🎯 Purpose & Impact
- 🛡️ Avoids invalid tracker states caused by zero-height or otherwise degenerate detections.
- 🎯 Improves tracking robustness when predictions fall entirely outside image boundaries and are clamped.
- 🔒 Limits the change to detection filtering before track creation, preserving existing confidence-threshold behavior.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @JESUSROYETH, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This automated message helps with the initial review, and an Ultralytics engineer will assist with the next steps. Please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Clean PR. The validity mask is applied consistently to both confidence partitions, preserves full-detection indices through the existing mask-based remapping, and prevents invalid dimensions from reaching `STrack.tlwh_to_xyah()`.

**Comment by glenn-jocher** [comment]
The filtering location and index preservation look correct, and CI is passing. Please add a focused regression test covering zero/negative dimensions in both confidence partitions and asserting that valid detections retain their original `idx`; line coverage alone does not protect this behavior from regression.
